core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -519,9 +510,11 @@ def _fit_folds(self,
             save_folds=save_folds, num_cpus=num_cpus, num_gpus=num_gpus
         )
         # noinspection PyCallingNonCallable
-        if fold_fitting_strategy == ParallelLocalFoldFittingStrategy:
+        if issubclass(fold_fitting_strategy, ParallelFoldFittingStrategy):
             fold_fitting_strategy_args['num_jobs'] = num_folds
             fold_fitting_strategy_args['num_folds_parallel'] = num_folds_parallel
+        if fold_fitting_strategy == ParallelDistributedFoldFittingStrategy:
+            fold_fitting_strategy_args['model_sync_path'] = os.environ.get("AG_MODEL_SYNC_PATH")
```

Comment:
Don't sneak in os environment variables in random places. All distributed env variable fetching should be in a singular place.

For example (please consider thinking of a better name, and I didn't actually test if the code is correct):
```

class DistributedVariables:
     @staticmethod
     @property
     def model_sync_path() -> str:
           return os.environ.get("AG_MODEL_SYNC_PATH")

     @staticmethod
     @property
     def is_distributed_mode() -> bool:
           return os.environ.get("AG_DISTRIBUTED_MODE", False)

```


```

fold_fitting_strategy_args['model_sync_path'] = DistributedVariables.model_sync_path
is_distributed_mode = DistributedVariables.is_distributed_mode

```